va/va.h

```diff
@@ -4225,6 +4225,62 @@ VAStatus vaDeriveImage (
     VAImage *image	/* out */
 );
 
+typedef struct _VASurfaceAllocationInfo{
+    /* format fourcc */
+    uint32_t    fourcc;
+    /* picture width x height */
+    uint16_t	width; 
+    uint16_t	height;
+    /* suggest data_size, it alwsys is minimum requirement*/ 
+    uint32_t	data_size;
+    /* number of planes , can not be greater than 3 */
+    uint32_t	num_planes;
+    /* 
+     * An array indicating the scanline pitch in bytes for each plane.
+     * Each plane may have a different pitch. Maximum 3 planes for planar formats
+     */
+    uint32_t	pitches[4];
+    /* 
+     * An array indicating the byte offset from the beginning of the image data 
+     * to the start of each plane.
+     */
+    uint32_t	offsets[4];
+    /*
+     * An array of ascii characters describing the order of the components within the bytes.
+     * Only entry_bytes characters of the string are used.by default , the order is from format fourcc
+     * if there are different layout, can use this value to describe it.
+     */
+    int8_t component_order[4];
+    /* base address alignment*/
+    uint32_t    base_address_align;
+    /* memory size alignment */
+    uint32_t    size_align;
+    /* suggest memory flags */
+    union{
+        struct{
+        uint32_t islinear :1;
+        uint32_t reserved :3;
+        }bits;
+        uint32_t flags;
+    };
+    /** \brief Reserved bytes for future use, must be zero */   
+    uint32_t                va_reserved[VA_PADDING_MEDIUM];
+
+}VASurfaceAllocationInfo;
+
+/** get the real driver surface information
+ * it can be a reference when call vaCreateSurface
+ * with system memory.such as user ptr */
+VAStatus vaQuerySurfaceAllocation(
+    VADisplay           dpy,
```

Comment:
Please, add all parameters to the doxygen info like:
`@param[in] dpy               the VA display`
Mark input/output parameters.